### PR TITLE
POTEL 21 - Drop OpenTelemetry spans for internal Sentry requests

### PR DESCRIPTION
**Pull request by adinauer**
#skip-changelog

## :scroll: Description

Requests to Sentry cause OpenTelemetry auto instrumentation to create spans. We now drop them in `SentrySampler` using `SamplingDecision.DROP`.

## :bulb: Motivation and Context

https://github.com/getsentry/sentry-java/issues/3499

## :pencil: Checklist

- [ ] I reviewed the submitted code.
- [ ] I added tests to verify the changes.
- [ ] No new PII added or SDK only sends newly added PII if `sendDefaultPII` is enabled.
- [ ] I updated the docs if needed.
- [ ] Review from the native team if needed.
- [ ] No breaking change or entry added to the changelog.
- [ ] No breaking change for hybrid SDKs or communicated to hybrid SDKs.

### CODE REVIEW

File: sentry-opentelemetry/sentry-opentelemetry-core/src/main/java/io/sentry/opentelemetry/OtelInternalSpanDetectionUtil.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    final @NotNull String spotlightUrl =
+        optionsSpotlightUrl != null ? optionsSpotlightUrl : "http://localhost:8969/stream";
+
+    if (containsSpotlightUrl(fullUrl, spotlightUrl)) {
+      return true;
+    }
+    if (containsSpotlightUrl(httpUrl, spotlightUrl)) {
+      return true;
+    }
+    //    }
```
Comment (lbloder) [inline sentry-opentelemetry/sentry-opentelemetry-core/src/main/java/io/sentry/opentelemetry/OtelInternalSpanDetectionUtil.java:53]: Leftover that can be deleted

File: sentry-opentelemetry/sentry-opentelemetry-core/src/main/java/io/sentry/opentelemetry/SentrySpanExporter.java [hunk 2]
```diff
@@ -134,51 +129,9 @@ private boolean isSpanTooOld(final @NotNull SpanData span, final @NotNull Sentry
   }
 
   private @NotNull List<SpanData> filterOutSentrySpans(final @NotNull Collection<SpanData> spans) {
```
Comment (lbloder) [inline sentry-opentelemetry/sentry-opentelemetry-core/src/main/java/io/sentry/opentelemetry/SentrySpanExporter.java:131]: Are there instances where `SentrySampler` is not invoked and spans need to be filtered here in the exporter?

Comment (adinauer) [reply sentry-opentelemetry/sentry-opentelemetry-core/src/main/java/io/sentry/opentelemetry/SentrySpanExporter.java:131]: Theoretically it's possible the span is modified during runtime and then has different attributes in `SpanExporter`.

File: sentry-opentelemetry/sentry-opentelemetry-core/src/main/java/io/sentry/opentelemetry/OtelInternalSpanDetectionUtil.java (hunk 1 above)
Comment (adinauer) [reply sentry-opentelemetry/sentry-opentelemetry-core/src/main/java/io/sentry/opentelemetry/OtelInternalSpanDetectionUtil.java:53]: Ideally we reintroduce the `if` above, then it's needed again. We can retest after doing the alpha release.

**Pull request merged**